warehouse/query-core/src/main/java/datawave/query/jexl/functions/TLDEventFieldAggregator.java

```diff
@@ -1,34 +1,41 @@
 package datawave.query.jexl.functions;
 
-import java.util.ArrayList;
-
 import org.apache.accumulo.core.data.ArrayByteSequence;
 import org.apache.accumulo.core.data.ByteSequence;
 
 import datawave.data.hash.UID;
 import datawave.query.predicate.EventDataQueryFilter;
-import datawave.query.tld.TLD;
 import datawave.query.util.TypeMetadata;
 
 public class TLDEventFieldAggregator extends EventFieldAggregator {
     public TLDEventFieldAggregator(String field, EventDataQueryFilter filter, int maxNextCount, TypeMetadata typeMetadata, String defaultTypeClass) {
         super(field, filter, maxNextCount, typeMetadata, defaultTypeClass);
     }
 
+    /**
+     * Event Data Key Structure (row, cf='datatype\0uid', cq='field\0value')
+     *
+     * @param cf
+     *            - a ByteSequence representing the Key's ColumnFamily
+     */
     @Override
-    protected ByteSequence parsePointer(ByteSequence columnFamily) {
-        // find the null between the dataType and Uid
-        ArrayList<Integer> nulls = TLD.instancesOf(0, columnFamily, 1);
-        final int start = nulls.get(0) + 1;
-        // uid is from the null byte to the end of the cf
-        ByteSequence uidByteSequence = columnFamily.subSequence(start, columnFamily.length());
-
-        UID uid = UID.parseBase(uidByteSequence.toString());
-        if (uid.getBaseUid().length() != uidByteSequence.length()) {
-            return new ArrayByteSequence(columnFamily.subSequence(0, start) + uid.getBaseUid());
+    protected ByteSequence parsePointer(ByteSequence cf) {
+        int index = 0;
```

Comment:
The TLD utility already has a method for `parseRootPointerFromFI(ByteSequence bytes)`, I think we should add a new method `parseRootPointerFromEvent(ByteSequence bytes)` and call that here.

Reply:
done with the name "parseDatatypeAndRootUidFromEvent" as we also need the datatype, not just the root uid